### DISCUSSION TITLE
footprints and tire tracks in snow and desert ？

### DISCUSSION THREAD

**Discussion by Smalldy** [TokisanGames/Terrain3D#555] [Questions] [answered]
I would like to ask if there are some recommended and relatively mature methods or best practices in the Godot Terrain3D plugin to achieve the effects of footprints and tire tracks in snow and desert? Are there any related sample codes, tutorial resources or plugin extensions for reference? I'm really looking forward to your reply and guidance, which will be of great help to my project development.

**Comment by TokisanGames** [comment, marked as answer]
A quick web search reveals plenty of code, shaders, and tutorials showing different methods for doing this. e.g. These and more. I didn't look for the best way. You will need to do that depending on your needs. There are many ways.
https://forum.godotengine.org/t/godot-4-footsteps-in-snow-possible/1694/2
https://godotshaders.com/shader/car-tracks-on-snow-or-sand-using-viewport-textures-and-particles/
https://youtu.be/rw8xqapk6as?si=gyO4CDh9_eHXDjqW

Some others in our discord have already done this for their games. You could search there as well.
